File: tests/shellcode/test_node_version.rs
use super::shell::Shell;
use super::{Call, ExpectCommandOutput};

pub(crate) fn test_node_version<S: Shell>(
    expected_version: &'static str,
) -> ExpectCommandOutput<S, Call> {
    ExpectCommandOutput::new(
        Call::new("node", vec!["-v"]),
        expected_version,
        "Node version",
    )
}

File: e2e/shellcode/test-node-version.ts
import { HasCall } from "./shells/cmdCall.js"
import { ScriptLine } from "./shells/types.js"
import { HasExpectCommandOutput } from "./shells/expect-command-output.js"

export default function testNodeVersion<
  S extends HasCall & HasExpectCommandOutput
>(shell: S, version: string): ScriptLine {
  const nodeVersion = shell.call("node", ["--version"])
  return shell.hasCommandOutput(nodeVersion, version, "node version")
}
